libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/distanceinfo/Gate.kt

```diff
@@ -0,0 +1,64 @@
+package com.mapbox.navigation.base.trip.model.roadobject.distanceinfo
+
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObjectPosition
+
+/**
+ * Gate represents information about a particular exit or entrance.
+ *
+ * @param id of the [Gate], persistent to the same gate.
+ * @param position on the road graph with coordinates
+ * @param probability to enter/exit this gate, value in range [0, 1].
+ * Warning: currently this field contains 1.0 for all gates, would be improved in the future.
+ * @param distance distance to the gate in meters:
+ * - if represents entrance outside of the object - positive
+ * - if represents entrance inside the object - negative
+ * - if represents exit outside the object - zero
+ * - if represents exit inside the object - positive
+ */
+class Gate internal constructor(
```

Comment:
Is it generated code? Might be `data class` if not

Reply:
This is how we implement builders internally to avoid breaking SemVer changes.